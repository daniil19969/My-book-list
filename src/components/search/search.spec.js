import React from "react";
import SearchTitle from "./search-title";
import SearchAuthor from "./search-author";
import SearchGenre from "./search-genre";
describe("Search component", () => {

it("should render component SearchTitle", () => {
    const mockFn = jest.fn();
    const input = shallow(<SearchTitle placeholder=""  onSearch={mockFn}/>);
    input.find('input').simulate('change', {target: {value: 'matched'} });
    expect(mockFn).toBeCalledWith(value);
});

it("should render component SearchAuthor", () => {
    const mockFn = jest.fn();
    const input = shallow(<SearchAuthor placeholder=""  onSearch={mockFn}/>);
    input.find('input').simulate('change', {target: {value: 'matched'} });
    expect(mockFn).toBeCalledWith(value);
});

it("should render component SearchGenre", () => {
    const mockFn = jest.fn();
    const input = shallow(<SearchGenre placeholder=""  onSearch={mockFn}/>);
    input.find('input').simulate('change', {target: {value: 'matched'} });
    expect(mockFn).toBeCalledWith(value);
});

});