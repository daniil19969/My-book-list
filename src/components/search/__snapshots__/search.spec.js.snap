// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render component SearchAuthor 1`] = `
<th>
  <input
    onChange={[Function]}
    placeholder="Автор"
    type="text"
  />
</th>
`;

exports[`Search component should render component SearchGenre 1`] = `
<th>
  <input
    onChange={[Function]}
    placeholder="Жанр"
    type="text"
  />
</th>
`;

exports[`Search component should render component SearchTitle 1`] = `
<th>
  <input
    onChange={[Function]}
    placeholder="Название"
    type="text"
  />
</th>
`;
